labelbox/data/serialization/coco/instance_dataset.py

```diff
@@ -0,0 +1,183 @@
+                                category_id=category_id,
+                                segmentation=[segmentation],
+                                area=shapely.area,
+                                bbox=[xmin, ymin, xmax - xmin, ymax - ymin],
+                                iscrowd=0)
+
+
+def rle_to_common(class_annotations, class_name):
+    mask = rle_decoding(class_annotations.segmentation.counts,
+                        *class_annotations.segmentation.size[::-1])
+    return ObjectAnnotation(name=class_name,
+                            value=Mask(mask=MaskData.from_2D_arr(mask),
+                                       color=[1, 1, 1]))
+
+
+def segmentations_to_common(class_annotations, class_name):
+    # Technically it is polygons. But the key in coco is called segmentations..
+    annotations = []
+    for points in class_annotations.segmentation:
+        annotations.append(
+            ObjectAnnotation(name=class_name,
+                             value=Polygon(points=[
+                                 Point(x=points[i], y=points[i + 1])
+                                 for i in range(0, len(points), 2)
+                             ])))
+    return annotations
+
+
+def process_label(label: Label,
+                  idx,
+                  image_root,
+                  max_annotations_per_image=10000):
+    annot_idx = idx * max_annotations_per_image
+    image_id = get_image_id(label, idx)
+    image = get_image(label, image_root, image_id)
+    coco_annotations = []
+    annotation_lookup = get_annotation_lookup(label.annotations)
+    categories = {}
+    for class_name in annotation_lookup:
+        for annotation in annotation_lookup[class_name]:
+            if annotation.name not in categories:
+                categories[annotation.name] = hash_category_name(
+                    annotation.name)
+            if isinstance(annotation.value, Mask):
+                coco_annotations.append(
+                    mask_to_coco_object_annotation(annotation, annot_idx,
+                                                   image_id,
+                                                   categories[annotation.name]))
+            elif isinstance(annotation.value, (Polygon, Rectangle)):
+                coco_annotations.append(
+                    vector_to_coco_object_annotation(
+                        annotation, annot_idx, image_id,
+                        categories[annotation.name]))
+            annot_idx += 1
+    return image, coco_annotations, categories
+
+
+class CocoInstanceDataset(BaseModel):
+    info: Dict[str, Any] = {}
+    images: List[CocoImage]
+    annotations: List[COCOObjectAnnotation]
+    categories: List[Categories]
+
+    @classmethod
+    def from_common(cls, labels: LabelCollection, image_root):
+        all_coco_annotations = []
+        categories = {}
+        images = []
+        futures = []
+        coco_categories = {}
+
+        with ProcessPoolExecutor(max_workers=8) as exc:
```

Comment:
let's make executor an optional argument of `from_common`

Reply:
See question below for same comment on panoptic dataset.